includes/legacy/class-wc-gateway-amazon-payments-advanced-subscriptions-legacy.php

```diff
@@ -384,13 +384,13 @@ private function get_billing_agreement_details( $order_id, $amazon_billing_agree
 		} catch ( Exception $e ) {
 			wc_apa()->log( "Error: Exception encountered in 'GetBillingAgreementDetails': {$e->getMessage()}" );
 
-			$return = new WP_Error();
-			$return->add(
+			$return = new WP_Error(
 				'billing_agreemment_details_failed',
 				is_object( $response ) && ! empty( $response->Error ) && is_object( $response->Error->Message ) && ! empty( $response->Error->Message ) ?
 				$response->Error->Message :
 				sprintf( __( 'Amazon API responded with an unexpected error when requesting for "GetBillingAgreementDetails" of billing agreement with ID %s', 'woocommerce-gateway-amazon-payments-advanced' ), $amazon_billing_agreement_id )
 			);
+			return $return;
```

Comment:
You can just return without the variable here.

Reply:
Should be ok now @msaggiorato, returning directly a new instance of WP_Error